gateway-service-impl/src/main/java/org/hypertrace/gateway/service/baseline/BaselineCalculator.java

```diff
@@ -1,33 +1,54 @@
 package org.hypertrace.gateway.service.baseline;
 
-import org.apache.commons.math.stat.descriptive.moment.Mean;
+import com.google.common.base.Preconditions;
 import org.apache.commons.math.stat.descriptive.moment.StandardDeviation;
 import org.apache.commons.math.stat.descriptive.rank.Median;
 import org.hypertrace.gateway.service.v1.baseline.Baseline;
 import org.hypertrace.gateway.service.v1.common.Value;
 import org.hypertrace.gateway.service.v1.common.ValueType;
 
+import java.util.List;
+
 public class BaselineCalculator {
 
-  public static Baseline getBaseline(double[] metricValueArray) {
-    Mean mean = new Mean();
+  public static Baseline getBaseline(List<Value> metricValues) {
+    double[] values = getValuesInDouble(metricValues);
+    return getBaseline(values);
+  }
+
+  private static Baseline getBaseline(double[] metricValueArray) {
+    Median median = new Median();
     StandardDeviation standardDeviation = new StandardDeviation();
-    double meanValue = mean.evaluate(metricValueArray);
+    double medianValue = median.evaluate(metricValueArray);
     double sd = standardDeviation.evaluate(metricValueArray);
-    double lowerBound = meanValue - (2 * sd);
+    double lowerBound = medianValue - (2 * sd);
     if (lowerBound < 0) {
       lowerBound = 0;
     }
-    double upperBound = meanValue + (2 * sd);
+    double upperBound = medianValue + (2 * sd);
     Baseline baseline =
         Baseline.newBuilder()
             .setLowerBound(
                 Value.newBuilder().setValueType(ValueType.DOUBLE).setDouble(lowerBound).build())
             .setUpperBound(
                 Value.newBuilder().setValueType(ValueType.DOUBLE).setDouble(upperBound).build())
             .setValue(
-                Value.newBuilder().setValueType(ValueType.DOUBLE).setDouble(meanValue).build())
+                Value.newBuilder().setValueType(ValueType.DOUBLE).setDouble(medianValue).build())
             .build();
     return baseline;
   }
+
+  private static double[] getValuesInDouble(List<Value> metricValues) {
+    Preconditions.checkArgument(metricValues.size() > 0,
```

Comment:
Does this mean we return an error response if I ask for the baseline and there's no historical data? That doesn't match how we treat empty data elsewhere.

Reply:
Fixed it.